drivers/entropy/entropy_stm32.c

```diff
@@ -30,6 +30,11 @@
 #define IRQN		DT_INST_IRQN(0)
 #define IRQ_PRIO	DT_INST_IRQ(0, priority)
 
+#if defined(CONFIG_SOC_SERIES_STM32L5X) || \
```

Comment:
Since the last push I am using `#if defined(RNG_CR_CONDRST)`, but now I think it would probably be even better to introduce new rng_v3 bindings instead.
Then we could add new properties which include the `RNG_HTCR HTCFG` initialization value and could possibly include a property to disable NIST compliance, set `CLKDIV`, and `RNG_CONFIGX` values.
Should we leave the condition on `RNG_CR_CONDRST` for now and introduce such new bindings in a future PR?

Reply:
I am not fully convinced about DT in this case: in all cases , driver need to implement former rng and rev 3.x .
Knowing that there is all we need in 
* CMSIS : RNG_CR_CONDRST which could eventually be replaced by (RNG_VER_3_2 || RNG_VER_3_1 || RNG_VER_3_0) to have explicit reference to REV3
* HAL : RNG_HTCFG  and RNG_HTCFG_1

So such information in DT would be redundant (adding complexity and risk of mistakes).
Other configurations like "disable NIST compliance, set CLKDIV, and RNG_CONFIGX values" are more optional and probably concerns only few experts. So I would leave it for further improvements. 

But let's wait @erwango feedback.

Reply:
I do agree with @ABOSTM on this.
This configuration options are quite specific, alternative configurations are mentioned as not tested (so "use at your own risk"), with no benefit documented and not certifiable.
So I don't think we should get into that "complexity" today, without a clearly established use case and expressed need.

Reply:
I didn't know whether there would be any good reason to choose any different configuration, but I do now agree that this ideally should simply work in a standard compliant configuration.
If you are fine with it, I'd leave it with `#if defined(RNG_CR_CONDRST)`